src/main/java/net/minecraftforge/common/config/ConfigManager.java

```diff
@@ -202,17 +230,52 @@ private static void createConfig(String modid, String category, Configuration cf
         {
             if (category.isEmpty())
                 throw new RuntimeException("Can not specify a primitive field when the category is empty: " + f.getDeclaringClass() +"/" + f.getName());
-            prop = adapter.getProp(cfg, category, f, instance, comment);
-            set(instance, f, adapter.getValue(prop));
+            if(!loading)
+            {
+                Property property = cfg.getCategory(category).get(getName(f));
+                Object fieldValue = get(instance, f);
+                Object propValue = adapter.getValue(property);
+                if(shouldReadFromVar(property, propValue, fieldValue)) 
+                {
+                    adapter.setProperty(cfg, category, f, fieldValue);
+                } 
+                else
+                {
+                    set(instance, f, propValue);
+                }
+                return;
```

Comment:
This is jut a nit-picky code style thing, but if (cond) not if(cond) and for one line if statements no need for braces.